### DISCUSSION TITLE
Inaccurate Tide Forecast Predictions

### DISCUSSION THREAD

**Discussion by mason-ford** [pyTMD/pyTMD#284] [Q&A]
Hello,
I'm eager to learn the library however I'm getting very inaccurate tide forecasts and I'm not sure what to do next. 
I started with the included "Plot Tide Forecasts.ipynb" notebook, downloaded multiple different models, ran the script and the results are always wrong. I'm getting tides that oscillate around 10-10.5 meters and the times are off. I must be doing something wrong.
Here is my result for La Jolla on 2024-02-23 using GOT4.10, I did not touch the included code.
![La Jolla 2024-02-23 GOT4 10](https://github.com/tsutterley/pyTMD/assets/33076236/48f836e8-ab4b-42b1-bf00-3591fcef6859)
In table format:
Tide Model: GOT
High Tides
Time: 9:20.0 height: 1055.628037
Time: 23:9.0 height: 1141.069106
Low Tide
Time: 4:28.0 height: 1023.753982
Time: 15:17.0 height: 999.007741

These are the tides for La Jolla from the NOAA website.
![image](https://github.com/tsutterley/pyTMD/assets/33076236/bd3b3c12-229c-4d6d-897d-0c43713bcf40)

I was hoping for a nudge in the right direction! I'd really like to use the library.
Thanks for any help

**Comment by mason-ford** [comment]
I realize that one is in LST and the other in UTC, but they are still off. 

The first problem I noticed is that the GOT model returns invalid harmonic constants for the LAT, and LONG given in the example. 
![image](https://github.com/tsutterley/pyTMD/assets/33076236/403660f6-9306-4792-aaa7-3e610dcb3b8c)

If you move the marker further out into the ocean it gives valid constants.
![image](https://github.com/tsutterley/pyTMD/assets/33076236/152c4198-7e90-4488-9ecc-6a86b09338a2)

I realized this using the "Solve Synthetic Tides" notebook.

Using valid harmonic constants will give more accurate tide information, however, the displacement or heights still seem to be off, only ever going between -150cm and 150cm. 

Is there an offset you have to apply?

**Comment by tsutterley** [comment]
Hey @mason-ford,
There can definitely be weird values where the program needs to extrapolate a model out to the coast (as you pointed out). Also with some of the global models, there may be some inaccuracies in the predictions on continental shelves. Except for these two notebooks (forecast and solve), everything should be in meters. You shouldn't need to scale anything to get the model predicted values for tide heights. I'm still spinning back up work wise after being away. I'll look into your problem on Monday to see if there's anything _extra_ weird going on.
